### PR TITLE
Implement update user grants func

### PR DESCRIPTION
**Pull request by wiseaidev**
## PR Acceptance Criteria
Related to #50

## Guidelines
* [X] PR follows the [Contributing guidelines](https://github.com/nylas-samples/nylas-hacktoberfest-2023/blob/main/CONTRIBUTION.md)?
* [X] PR Adheres to our [Code of Conduct](https://github.com/nylas-samples/nylas-hacktoberfest-2023/blob/main/CODE_OF_CONDUCT.md)?
* [X] Have you checked to ensure there aren't other open Pull Requests for the same update/change?

## Code complete
* [X] A `README.md` available describing how to setup and test the code?
* [X] The code sample in the correct location (i.e. `grant/python/update_grants`)?
* [x] Link added to the repository readme (view `# Code Snippets created by the Nylas Community` with examples) describing what the code sample does for others to try out
* [x] The code can be run using Github Codespaces, so the maintainer can try out the code sample remotely to ensure everything is working before we accept the PR.

## For PR Reviewer use
* [x] Purpose of code sample is clear
* [x] The code sample is readable
* [x] Steps try out the code are provided
* [x] Code sample runs on codespaces environment
* [x] Code works as expected
* [x] PR labels added

### CODE REVIEW

File: code-samples/grant/python/update_grants/update_grants/__main__.py
Comment (relaxedtomato) [reply code-samples/grant/python/update_grants/update_grants/__main__.py:58]: @wiseaidev - can you share more details?

- [ ] What error are you seeing?
- [ ] Can you share a screenshot? Or logs?

Happy to take a look and help resolve, and we'll help progress the PR for hacktoberfest!

Comment (wiseaidev) [reply code-samples/grant/python/update_grants/update_grants/__main__.py:58]: Have a look at the readme file; The `update` wasn't successful (a bug in the sdk/api). The error message isn't helpful, unfortunately - needs debugging:

```sh
    *** Running the app locally... ***
    poetry run update_grants
    User Grants Before Update:

    Grant ID: 80c042c2-8c30-4037-bd41-d4e54c8220c6, Status: mahmoudddharmouchhh%40gmail.com%3Agoogle
    An error occurred: Expecting value: line 1 column 1 (char 0) # <------------ Error occurred during the update 

    User Grants After Update:
    Grant ID: 80c042c2-8c30-4037-bd41-d4e54c8220c6, Status: mahmoudddharmouchhh%40gmail.com%3Agoogle
```

File: code-samples/grant/python/update_grants/update_grants/__main__.py
Comment (relaxedtomato) [inline code-samples/grant/python/update_grants/update_grants/__main__.py:88]: @wiseaidev - recommend to leave this as something like:
`grant_id_to_update = ADD_GRANT_ID_TO_UPDATE`

File: code-samples/grant/python/update_grants/README.md
Comment (relaxedtomato) [inline code-samples/grant/python/update_grants/README.md:74]: Same for these, remove any user specific data before we merge.

File: code-samples/grant/python/update_grants/README.md
Comment (relaxedtomato) [inline code-samples/grant/python/update_grants/README.md:78]: Same for these, remove any user specific data before we merge.

File: code-samples/grant/python/update_grants/update_grants/__main__.py
Comment (relaxedtomato) [reply code-samples/grant/python/update_grants/update_grants/__main__.py:58]: I will revisit this bug, however, this will not block the PR from being accepted, we will accept the PR once any other comments are addressed.

**Review by relaxedtomato — CHANGES_REQUESTED** [review]
Small changes requested, this looks good - we'll take a look at the bug outside of this PR and market it as `hacktoberfest-approved` after the changes are made!

File: code-samples/grant/python/update_grants/README.md
Comment (wiseaidev) [reply code-samples/grant/python/update_grants/README.md]: Done!

File: code-samples/grant/python/update_grants/update_grants/__main__.py
Comment (wiseaidev) [reply code-samples/grant/python/update_grants/update_grants/__main__.py]: Done!

File: code-samples/grant/python/update_grants/README.md
Comment (wiseaidev) [reply code-samples/grant/python/update_grants/README.md]: Done!

**Review by atejada — APPROVED** [review]
All good 👍

**Review by relaxedtomato — APPROVED** [review]
looks solid @wiseaidev !

**Pull request merged**
